cleanlab/object_detection/filter.py

```diff
@@ -217,3 +234,146 @@ def _pool_box_scores_per_image(is_issue_per_box: List[np.ndarray]) -> np.ndarray
         if np.sum(issue_per_box) > 0:
             is_issue[idx] = 1
     return is_issue
+
+
+def _process_class_list(class_list: List[np.ndarray], class_dict: Dict[int, float]):
+    """
+    Converts a list of classes using a float dictionary into a list where each class is replaced by its corresponding float value.
+    """
+    class_l2 = []
+    for i in class_list:
+        l3 = [class_dict[j] for j in i]
+        class_l2.append(l3)
+    return class_l2
+
+
+def calculate_ap_per_class(
+    labels: List[Dict[str, Any]],
+    predictions: List[np.ndarray],
```

Comment:
This should be a private function `_claculate_ap_per_class()`. Also I suggest making iou_threshold and num_procs optional kwargs like so:

`_claculate_ap_per_class(labels, predictions, *, iou_threshold, num_procs)`

Reply:
https://github.com/cleanlab/cleanlab/blob/3024fad44f236c2f546610be35c3cfa5d011ce31/cleanlab/object_detection/filter.py#L250